signer/threshold_validator.go

```diff
@@ -369,20 +464,34 @@ func (pv *ThresholdValidator) LoadSignStateIfNecessary(chainID string) error {
 }
 
 func (pv *ThresholdValidator) getExistingBlockSignature(chainID string, block *Block) ([]byte, time.Time, error) {
-	height, round, step, stamp, signBytes := block.Height, block.Round, block.Step, block.Timestamp, block.SignBytes
-	hrs := HRSKey{
-		height,
-		round,
-		step,
-	}
-
 	css := pv.mustLoadChainState(chainID)
 
-	latestBlock, existingSignature := css.lastSignState.GetFromCache(
-		hrs,
-		css.lastSignStateMutex,
-	)
+	latestBlock, existingSignature := css.lastSignState.GetFromCache(block.HRSKey())
+
+	return pv.compareBlockSignature(chainID, block, latestBlock, existingSignature)
+}
+
+func (pv *ThresholdValidator) compareBlockSignature(
```

Comment:
Can you add godoc for this so other readers understand what the return values mean?